### PR TITLE
fix: bind account

### PR DESCRIPTION
**Pull request by LiusCraft**

### CODE REVIEW

File: internal/core/casdoor.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	if err != nil {
+		p.xLog.Error(err)
+		return map[string]bool{}
+	}
+	u, err := casdoorsdk.GetUser(claim.Name)
+	if err != nil {
+		return map[string]bool{}
+	}
+
+	return map[string]bool{
```
Comment (IRONICBo) [inline internal/core/casdoor.go]: Use struct instead.

File: internal/core/casdoor.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package core
```
Comment (IRONICBo) [inline internal/core/casdoor.go]: Why not move them to `user.go`?

File: internal/core/casdoor.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+func (p *Community) UnLink(token, provider string) bool {
+	claim, err := casdoorsdk.ParseJwtToken(token)
+	if err != nil {
+		p.xLog.Error(err)
+		return false
+	}
+	u, err := casdoorsdk.GetUser(claim.Name)
+	if err != nil {
+		log.Println(err)
```
Comment (IRONICBo) [inline internal/core/casdoor.go]: Use xlog.

File: internal/core/casdoor.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	if err != nil {
+		return false
+	}
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return false
+	}
+
+	if response.Status != "ok" {
+		log.Println(response.Msg)
```
Comment (IRONICBo) [inline internal/core/casdoor.go]: Use xlog.

File: internal/core/casdoor_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+func TestGetAccountBinds(t *testing.T) {
+	conf := &Config{}
+	todo := context.TODO()
+	community, err := New(todo, conf)
+	if err != nil {
+		t.Skip(err)
```
Comment (IRONICBo) [inline internal/core/casdoor_test.go]: Do not skip error.

File: internal/core/casdoor_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	}
+	t.Log(community.GetAccountBinds(token))
+}
+
+func TestAccountUnLink(t *testing.T) {
+	conf := &Config{}
+	todo := context.TODO()
+	community, err := New(todo, conf)
+	if err != nil {
+		t.Skip(err)
```
Comment (IRONICBo) [inline internal/core/casdoor_test.go]: Do not skip them.

**Review by IRONICBo — CHANGES_REQUESTED** [review]

File: cmd/gopcomm/community_yap.gox [hunk 7]
```diff
@@ -120,22 +121,43 @@ get "/user/:id", ctx => {
...
+get "/userEdit", ctx => {
+	token, err := core.GetToken(ctx)
 	if err != nil {
 		http.Redirect(ctx.ResponseWriter, ctx.Request, "/error", http.StatusTemporaryRedirect)
 	}
+	gac, err := gopaccountsdk.GetClient(token.Value)
+	if err != nil {
+		http.Redirect(ctx.ResponseWriter, ctx.Request, "/error", http.StatusTemporaryRedirect)
+	}
+	println gac
```
Comment (IRONICBo) [inline cmd/gopcomm/community_yap.gox]: Suggest to remove it.

File: cmd/gopcomm/community_yap.gox [hunk 8]
```diff
@@ -120,22 +121,43 @@ get "/user/:id", ctx => {
 		"UserId":      user.Id,
 	}
 }
-
-get "/userEdit", ctx => {
-	var user *core.User
+get "/userUnlink", ctx => {
```
Comment (IRONICBo) [inline cmd/gopcomm/community_yap.gox:124]: Suggest to use `/user/unlink` instead.

File: cmd/gopcomm/community_yap.gox [hunk 9]
```diff
@@ -120,22 +121,43 @@ get "/user/:id", ctx => {
...
+		http.Redirect(ctx.ResponseWriter, ctx.Request, "/userEdit", http.StatusTemporaryRedirect)
 	}
-
+	gac, err := gopaccountsdk.GetClient(token.Value)
+	if err == nil {
+		gac.UnLink(pv)
+	}
+	http.Redirect(ctx.ResponseWriter, ctx.Request, "/userEdit", http.StatusTemporaryRedirect)
+}
+get "/userEdit", ctx => {
```
Comment (IRONICBo) [inline cmd/gopcomm/community_yap.gox:145]: Ditto

File: cmd/gopcomm/community_yap.gox (hunk 8 above)
Comment (LiusCraft) [reply cmd/gopcomm/community_yap.gox:124]: Yap now support this type of router?

File: cmd/gopcomm/community_yap.gox [hunk 10]
```diff
@@ -120,22 +121,43 @@ get "/user/:id", ctx => {
...
-		user, err = community.GetUser(token.Value)
-		if err != nil {
-			xLog.Error("get user error:", err)
-		}
+		gac.UnLink(pv)
 	}
-
+	http.Redirect(ctx.ResponseWriter, ctx.Request, "/userEdit", http.StatusTemporaryRedirect)
+}
+get "/userEdit", ctx => {
```
Comment (IRONICBo) [inline cmd/gopcomm/community_yap.gox:145]: `get /page/user-info`

**Review by IRONICBo — APPROVED** [review]
LGTM

**Pull request merged**